### PR TITLE
doc: Small doc and example improvements for `mongodbatlas_privatelink_endpoint_service`

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Small doc and example improvements for `mongodbatlas_privatelink_endpoint_service`

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals I have added appropriate changelog entries.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Comment by lantoli** [comment]
improve example for https://github.com/mongodb/terraform-provider-mongodbatlas/issues/2782

File: docs/resources/advanced_cluster.md [hunk 1]
```diff
@@ -661,8 +661,7 @@ In addition to all arguments above, the following attributes are exported:
...
 
-   Private connection strings are not available until the respective `mongodbatlas_privatelink_endpoint_service` resources are fully applied. Add a `depends_on = [mongodbatlas_privatelink_endpoint_service.example]` to ensure `connection_strings` are available following `terraform apply`.\
-   If the expected connection string(s) do not contain a value, a `terraform refresh` may need to be performed to obtain the value. One can also view the status of the peered connection in the [Atlas UI](https://docs.atlas.mongodb.com/security-vpc-peering/).
+   Private connection strings are not available until the respective `mongodbatlas_privatelink_endpoint_service` resources are fully applied. Add a `depends_on = [mongodbatlas_privatelink_endpoint_service.example]` to ensure `connection_strings` are available following `terraform apply`. If the expected connection string(s) do not contain a value, a `terraform refresh` may need to be performed to obtain the value. One can also view the status of the peered connection in the [Atlas UI](https://docs.atlas.mongodb.com/security-vpc-peering/).
```
Comment (lantoli) [inline docs/resources/advanced_cluster.md:664]: delete \

File: examples/mongodbatlas_privatelink_endpoint/azure/main.tf [hunk 2]
```diff
@@ -59,6 +59,7 @@ data "mongodbatlas_advanced_cluster" "cluster" {
   count      = var.cluster_name == "" ? 0 : 1
   project_id = var.project_id
   name       = var.cluster_name
+  depends_on = [mongodbatlas_privatelink_endpoint_service.test]
```
Comment (lantoli) [inline examples/mongodbatlas_privatelink_endpoint/azure/main.tf:62]: add dependency as suggested in adv_cluster doc

**Review by cveticm — APPROVED** [review]
LGTM 👍

**Review by EspenAlbert — APPROVED** [review]
LGTM

**Pull request merged**